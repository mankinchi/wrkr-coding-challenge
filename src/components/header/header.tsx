import { Menu } from "react-feather";

interface Props {
	onMenuToggle: () => void;
}

export const Header = (props: Props) => {
	const { onMenuToggle } = props;

	return (
		<div className="flex h-12 items-center gap-2 border-b border-solid border-b-black px-2">
			<Menu
				size={20}
				className="block cursor-pointer md:hidden"
				onClick={onMenuToggle}
			/>

			<div className="text-2xl font-extrabold">Header</div>
		</div>
	);
};
