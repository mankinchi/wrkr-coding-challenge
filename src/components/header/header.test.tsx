import { fireEvent, render } from "@testing-library/react";
import { Header } from "./header";

describe("Header component", () => {
	it("should render correctly", () => {
		const { container } = render(<Header onMenuToggle={() => {}} />);
		expect(container).toMatchSnapshot();
	});

	it("should execute onMenuToggle when click toggle menu icon", () => {
		const onMenuToggle = jest.fn();

		const { container } = render(<Header onMenuToggle={onMenuToggle} />);

		expect(container.querySelector("svg")).not.toBeNull();

		fireEvent.click(container.querySelector("svg"));
		expect(onMenuToggle).toBeCalledTimes(1);
	});
});
