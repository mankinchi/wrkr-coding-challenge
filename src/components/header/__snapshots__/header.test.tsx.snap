// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <div
    class="flex h-12 items-center gap-2 border-b border-solid border-b-black px-2"
  >
    <svg
      class="block cursor-pointer md:hidden"
      data-cy="navigation-toggle-handler"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
    <div
      class="text-2xl font-extrabold"
    >
      Header
    </div>
  </div>
</div>
`;
